### PR TITLE
Check format with mill

### PR DESCRIPTION
**Pull request by sequencer**
reformat lit and panama and use mill to check format

### Contributor Checklist

- [ ] Did you add Scaladoc to every public function/method?
- [ ] Did you add at least one test demonstrating the PR?
- [ ] Did you delete any extraneous printlns/debugging code?
- [ ] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [ ] Did you request a desired merge strategy?
- [ ] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

- Internal or build-related (includes code refactoring/cleanup)

#### Desired Merge Strategy

- Squash

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels? (Select the most appropriate one based on the "Type of Improvement")
- [ ] Did you mark the proper milestone (Bug fix: `3.6.x`, `5.x`, or `6.x` depending on impact, API modification or big change: `7.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -130,14 +130,14 @@ jobs:
         uses: actions/checkout@v3
         with:
           ref: ${{ inputs.ref }}
-      - name: Setup Scala
-        uses: actions/setup-java@v3
+      - name: Install Mill
+        uses: jodersky/setup-mill@v0.3.0
```
Comment (jackkoenig) [inline .github/workflows/test.yml:134]: What version of Java is this using?

**Comment by jackkoenig** [comment]
So it doesn't return fail, but this crashes for me with:
```
error: /Users/koenig/work/chisel-optimize-bitpat/panamaom/src/PanamaCIRCTOM.scala: org.scalafmt.dynamic.exceptions.PositionExceptionImpl: /Users/koenig/work/
chisel-optimize-bitpat/panamaom/src/PanamaCIRCTOM.scala:120: error: ; expected but string constant found                                                     
      s"mlirString{\"$mlirString\"}"                                                                                                                         
                                 ^                                                                                                                           
Caused by: /Users/koenig/work/chisel-optimize-bitpat/panamaom/src/PanamaCIRCTOM.scala:120: error: ; expected but string constant found                       
      s"mlirString{\"$mlirString\"}"                                                                                                                         
                                 ^                                                                                                                           
        at scala.meta.internal.parsers.Reporter.syntaxError(Reporter.scala:16)                  
        at scala.meta.internal.parsers.Reporter.syntaxError$(Reporter.scala:16)                                                                              
        at scala.meta.internal.parsers.Reporter$$anon$1.syntaxError(Reporter.scala:22)                                                                       
        at scala.meta.internal.parsers.Reporter.syntaxError(Reporter.scala:17)                                                                               
        at scala.meta.internal.parsers.Reporter.syntaxError$(Reporter.scala:17)                                       
        at scala.meta.internal.parsers.Reporter$$anon$1.syntaxError(Reporter.scala:22)                                                                       
        ....
```
Which is a bit weird, only on the first run (i.e. if I delete `out` and run `mill __.checkFormat`)

**Comment by sequencer** [comment]
really strange. reproducible in Linux as well.

**Comment by sequencer** [comment]
Looks like the scalafmt version is too old, it can be parsed by scalafmt 3.8.1.

However it might need to reformat the world...

**Comment by sequencer** [comment]
An solution can be removing the check of panama for now...

**Comment by jackkoenig** [comment]
Alternatively we can tweak the Panama code to work around the issue, change `s"mlirString{\"$mlirString\"}"` to `s"""mlirString{"$mlirString"}"""`

But yeah

**Comment by sequencer** [comment]
> Alternatively we can tweak the Panama code to work around the issue, change s"mlirString{\"$mlirString\"}" to s"""mlirString{"$mlirString"}"""

Fixed...

**Review by jackkoenig — APPROVED** [review]

**Comment by sequencer** [comment]
Thanks Jack for updating the documentation!

**Pull request merged**